subprojects/docs/src/docs/release/notes.md

```diff
@@ -138,9 +140,18 @@ These methods have been deprecated and the `create()` or `register()` methods sh
 
 ## Potential breaking changes
 
-### Kotlin DSL breakages
+### Changes to the Gradle Kotlin DSL
 
-- `project.java.sourceSets` is now `project.sourceSets`
+The Kotlin DSL now enables the _SAM conversion for Kotlin functions_ Kotlin compiler feature in order to expose an uniform `org.gradle.api.Action<T>` based API to both Groovy DSL and Kotlin DSL scripts.
+
+The DSL types and behavior of containers elements delegated properties (e.g. `val jar by tasks`) and containers scopes (e.g. `tasks { }`) changed.
+
+The source set container can now be accessed using `project.sourceSets`, or just `sourceSets`.
+Previously it was located at `project.java.sourceSets`, or just `java.sourceSets`.  
+
+All these changes could cause script compilation errors.
+
+See the [release notes](https://github.com/gradle/kotlin-dsl/releases/tag/v1.0-RC1) for more information and how to fix builds broken by the changes described above.
```

Comment:
could mayyybe say "Kotlin DSL release notes" for the link. Kinda silly, but we see parts of release notes copied to blogs etc. and I weakly feel that we should fully qualify here.